src/plugins/discover/public/application/angular/discover.js

```diff
@@ -1085,62 +799,6 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
     });
   }
 
-  function getIndexPatternWarning(index) {
-    return i18n.translate('discover.valueIsNotConfiguredIndexPatternIDWarningTitle', {
-      defaultMessage: '{stateVal} is not a configured index pattern ID',
-      values: {
-        stateVal: `"${index}"`,
-      },
-    });
-  }
-
-  function resolveIndexPatternLoading() {
-    const {
-      loaded: loadedIndexPattern,
-      stateVal,
-      stateValFound,
-    } = $route.current.locals.savedObjects.ip;
-
-    const ownIndexPattern = $scope.searchSource.getOwnField('index');
-
-    if (ownIndexPattern && !stateVal) {
-      return ownIndexPattern;
-    }
-
-    if (stateVal && !stateValFound) {
-      const warningTitle = getIndexPatternWarning();
-
-      if (ownIndexPattern) {
-        toastNotifications.addWarning({
-          title: warningTitle,
-          text: i18n.translate('discover.showingSavedIndexPatternWarningDescription', {
-            defaultMessage:
-              'Showing the saved index pattern: "{ownIndexPatternTitle}" ({ownIndexPatternId})',
-            values: {
-              ownIndexPatternTitle: ownIndexPattern.title,
-              ownIndexPatternId: ownIndexPattern.id,
-            },
-          }),
-        });
-        return ownIndexPattern;
-      }
-
-      toastNotifications.addWarning({
-        title: warningTitle,
-        text: i18n.translate('discover.showingDefaultIndexPatternWarningDescription', {
-          defaultMessage:
-            'Showing the default index pattern: "{loadedIndexPatternTitle}" ({loadedIndexPatternId})',
-          values: {
-            loadedIndexPatternTitle: loadedIndexPattern.title,
-            loadedIndexPatternId: loadedIndexPattern.id,
-          },
-        }),
-      });
-    }
-
-    return loadedIndexPattern;
-  }
-
```

Comment:
@maja was extracted to 
https://github.com/elastic/kibana/blob/5a1f3ef6f740315b7b41013f840a36b48cd68c66/src/plugins/discover/public/application/helpers/resolve_index_pattern.ts#L108